File: frontend/src/types/profilePicture.ts
export interface ProfilePicture {
  id: string;
  title?: string;
  filename: string;
  original_path: string;
  variants: Record<string, ImageVariant>;
  is_active: boolean;
  file_size?: number;
  width?: number;
  height?: number;
  created_at: string;
  updated_at: string;

  // URL fields populated by API
  original_url?: string;
  download_url?: string;
}

export interface ImageVariant {
  path: string;
  width: number;
  height: number;
  size_bytes: number;
  format: string;
  url?: string;
}

export interface ProfilePictureCreate {
  title?: string;
  is_active?: boolean;
}

export interface ProfilePictureUpdate {
  title?: string;
  is_active?: boolean;
}

export interface ProfilePictureListResponse {
  profile_pictures: ProfilePicture[];
  total: number;
}

export interface ActiveProfilePictureResponse {
  profile_picture: ProfilePicture | null;
}

export interface ProfilePictureUploadData {
  file: File;
  title?: string;
}

File: backend/app/schemas/profile_picture.py
from __future__ import annotations

from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, field_validator


class ProfilePictureBase(BaseModel):
    title: str | None = None
    is_active: bool = False


class ProfilePictureCreate(ProfilePictureBase):
    pass


class ProfilePictureUpdate(BaseModel):
    title: str | None = None
    is_active: bool | None = None


class ProfilePictureResponse(ProfilePictureBase):
    id: str
    filename: str
    original_path: str
    variants: dict[str, Any]
    file_size: int | None = None
    width: int | None = None
    height: int | None = None
    created_at: datetime
    updated_at: datetime

    # URL fields (populated by API)
    original_url: str | None = None
    download_url: str | None = None

    @field_validator("id", mode="before")
    @classmethod
    def convert_uuid_to_string(cls, v: UUID | str) -> str:
        """Convert UUID to string for JSON serialization."""
        return str(v) if isinstance(v, UUID) else v

    class Config:
        from_attributes = True


class ProfilePictureListResponse(BaseModel):
    profile_pictures: list[ProfilePictureResponse]
    total: int


class ActiveProfilePictureResponse(BaseModel):
    profile_picture: ProfilePictureResponse | None = None
